src/Interpreters/InterpreterDropQuery.cpp

```diff
@@ -137,6 +142,11 @@ BlockIO InterpreterDropQuery::executeToTableImpl(ContextPtr context_, ASTDropQue
 
         if (query.kind == ASTDropQuery::Kind::Detach)
         {
+            /// proton: starts
+            if (table->isExternalTable())
+                throw Exception("Cannot DETACH external table", ErrorCodes::SYNTAX_ERROR);
```

Comment:
Let's use a different error code, like NOT_IMPLEMENTED, SUPPORTED instead of SYNTAX_ERROR